File: apps/tv/modules/valence-focus-fence/android/src/main/java/app/valence/tv/focusfence/ValenceFocusFenceModule.kt
package app.valence.tv.focusfence

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

/** Offers JavaScript a view that keeps the remote out of whatever it holds while it is shut. */
class ValenceFocusFenceModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("ValenceFocusFence")

    View(ValenceFocusFenceView::class) {
      Prop("isShut") { view: ValenceFocusFenceView, isShut: Boolean ->
        view.isShut = isShut
      }
    }
  }
}

File: apps/tv/modules/valence-focus-fence/ios/ValenceFocusFenceModule.swift
import ExpoModulesCore

/// Offers JavaScript a view that keeps the remote out of whatever it holds while it is shut.
public class ValenceFocusFenceModule: Module {
  public func definition() -> ModuleDefinition {
    Name("ValenceFocusFence")

    View(ValenceFocusFenceView.self) {
      Prop("isShut") { (view: ValenceFocusFenceView, isShut: Bool) in
        view.isShut = isShut
      }
    }
  }
}
